emcc.py

```diff
@@ -2495,6 +2496,10 @@ def phase_linker_setup(options, state, newargs):
       'removeRunDependency',
     ]
 
+  if options.embind_emit_tsd:
+    settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['$embindEmitTypes']
+    settings.REQUIRED_EXPORTS += ['free']
```

Comment:
Why `free` here?

Reply:
I get `cache:INFO:  - ok
error: undefined symbol: free (referenced by $getTypeName, referenced by $embindEmitTypes, referenced by root reference (e.g. compiled C/C++ code))` without it. I'm not sure how this is currently working with regular embind, since those functions already exist.

Reply:
`$getTypeName` has `$getTypeName__deps: ['$readLatin1String', '__getTypeName', 'free'],` Should that be enough to have free included or do we need to export it too?

Reply:
Yes that should be enough.

I wonder if you are running into this issue: https://reviews.llvm.org/D153466

Can you tell me how I can repro this?

Reply:
`test/runner other.test_embind_tsgen`

Reply:
Hmm.. I see the problem .. we are using `DEFAULT_LIBRARY_FUNCS_TO_INCLUDE` to included the library function, we don't give that information to the linker.   We can look into the fixing that but I think a simpler solution might be to move `requireRegisteredType` and `getTypeName` out of the shared embind file.   (I don't think they are needed for ts case, right?.. are there other functions too that don't need to be there maybe?)

Reply:
Oh wait .. getTypeName is totally used by embindEmitTypes

Reply:
How about you leave this `settings.REQUIRED_EXPORTS += ['free']` in this PR with TODO to remove it.  I'll open a bug about it.

Reply:
https://github.com/emscripten-core/emscripten/issues/19759